redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts

```diff
@@ -411,6 +416,122 @@ export class RecommendationProvider {
     }
   }
 
+  /**
+  * Check search indexes recommendation
+   * @param redisClient
+   * @param keys
+   * @param client
+   */
+  // eslint-disable-next-line
+  async determineSearchIndexesRecommendation(
+    redisClient: Redis,
+    keys: Key[],
+    client: any,
+  ): Promise<Recommendation> {
+    try {
+      if (client.isCluster) {
+        let processedKeysNumber = 0;
+        let isJSONOrHash = false;
+        let sortedSetNumber = 0;
+        while (
```

Comment:
while? why? 
I think we should use here Promise.all to not wait before processing the next key.
In general I think that this implementation should be the same as for Standalone databases but with Promise.all -> sendCommand instead of piepeline